io-sim-classes/src/Control/Monad/Class/MonadSTM.hs

```diff
@@ -45,6 +45,10 @@ module Control.Monad.Class.MonadSTM
   , isEmptyTBQueueDefault
   , isFullTBQueueDefault
   , lengthTBQueueDefault
+
```

Comment:
Which versions of the `stm` package do you intend to support? I don't you can be both a drop-in replacement for `stm` *and* support the entire `stm >=2.4 && <2.6` version range. That's because some types changed and some functions were added. See https://github.com/haskell/stm/blob/master/changelog.md

We had to use the following workaround for `lengthTBQueue` in consensus: https://github.com/input-output-hk/ouroboros-network/blob/d81d370008727ac9811bbc4fc70ea0cbcde7cd28/ouroboros-consensus/src/Ouroboros/Consensus/Util/IOLike.hs#L68-L71

Reply:
According to [libraries version history](https://gitlab.haskell.org/ghc/ghc/-/wikis/commentary/libraries/version-history), `ghc-8.6.5` to `ghc-8.10` support `stm-2.5.0`.  I think the right thing is to bump version of `monad-class` to `0.2.0.0` and drop support for `stm-2.4.x`. We could support both but it would be ugly (using CPP).